src/time.ts

```diff
@@ -1,5 +1,27 @@
+import { MaybePromise } from "./types"
+
 export const delay = (milliseconds: number) => {
   return new Promise<void>((resolve) => {
     setTimeout(resolve, milliseconds)
   })
 }
+
+export async function until(
+  cb: () => MaybePromise<boolean>,
+  interval: number,
+  maxRetries: number = Infinity,
+  timeoutMessage?: string,
+): Promise<void> {
+  let retryCount: number = 0
+
+  while (retryCount < maxRetries) {
+    const res = await cb()
+    if (res) {
+      return
```

Comment:
could it be that you expect some data to return?

Reply:
Not in the setup when callback returns boolean. 

Case when callback returns `Promise<T>` would be different, as it can fail for multiple reasons, I'd rather add it later if needed.

Reply:
could you please show the example of  implementation?
For some reason I feel that this is rather rarer case, when you don't expect some retriable to return any value, and most probably you would, even if it's boolean - it could be explicit false, (not an exception), in this case you'd still expect it to be returned.. sorry, may be i'm overthinking, that's why it'd be great to see implementation

Thanks!

Reply:
The use case is just "waiting for something to become true". 

I had it in tests, but I believe it could find use cases in other places where we need to poll async processes. 

```ts
test("Phase 2: cmd-bot start pipeline", async () => {
  const mockServers = ensureDefined(getMockServers())

  const mockedPipelineEndpoint = await mockServers.gitLab
    .forPost("/api/v4/projects/tripleightech%2Fcommand-bot-test/pipeline")
    .withQuery({ ref: "cmd-bot/4" })
    .thenReply(201, restFixures.gitlab.pendingPipeline, jsonResponseHeaders)

  await until(async () => !(await mockedPipelineEndpoint.isPending()), 100, 50)
})

test("Phase 3: cmd-bot updates the comment with a link to the pipeline", async () => {
  const comment = ensureDefined(commentThatBotLeft)

  await until(
    () => comment.body.includes("/cmd cancel"),
    100,
    50,
    "Expected bot to edit commit so it would include pipeline cancellation command." +
      `Comment body now is: ${comment.body}`,
  )
})
```

Reply:
ahh i see, make sense, i didn't pay attention that the first `cb` is actually `MaybePromise<boolean>`
👍